desktop/src/main/java/bisq/desktop/main/dao/burnbsq/burningman/BalanceEntryItem.java

```diff
@@ -0,0 +1,219 @@
+import java.util.Date;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+
+@EqualsAndHashCode
+class BalanceEntryItem {
+    private final BalanceEntry balanceEntry;
+    private final BsqFormatter bsqFormatter;
+    private final CoinFormatter btcFormatter;
+    @Getter
+    private final Date date;
+    @Getter
+    private final Date month;
+    @Getter
+    private final Optional<Price> price;
+    @Getter
+    private final Optional<BalanceEntry.Type> type;
+    @Getter
+    private final Optional<Long> receivedBtc;
+    @Getter
+    private final Optional<Long> receivedBtcAsBsq;
+    @Getter
+    private final Optional<Long> burnedBsq;
+    @Getter
+    private final Optional<Long> revenue;
+
+    // We create the strings on demand and cache them. For large data sets it would be a bit slow otherwise.
+    private String monthAsString, dateAsString, receivedBtcAsString, receivedBtcAsBsqAsString, burnedBsqAsString, revenueAsString,
+            priceAsString, typeAsString;
+
+    BalanceEntryItem(BalanceEntry balanceEntry,
+                     Map<Date, Price> averageBsqPriceByMonth,
+                     BsqFormatter bsqFormatter,
+                     CoinFormatter btcFormatter) {
+        this.balanceEntry = balanceEntry;
+        this.bsqFormatter = bsqFormatter;
+        this.btcFormatter = btcFormatter;
+
+        date = balanceEntry.getDate();
+        month = balanceEntry.getMonth();
+        price = Optional.ofNullable(averageBsqPriceByMonth.get(month));
+        if (balanceEntry instanceof MonthlyBalanceEntry) {
+            MonthlyBalanceEntry monthlyBalanceEntry = (MonthlyBalanceEntry) balanceEntry;
+            receivedBtc = Optional.of(monthlyBalanceEntry.getReceivedBtc());
+            burnedBsq = Optional.of(-monthlyBalanceEntry.getBurnedBsq());
+            type = monthlyBalanceEntry.getTypes().size() == 1 ?
+                    Optional.of(new ArrayList<>(monthlyBalanceEntry.getTypes()).get(0)) :
+                    Optional.empty();
+        } else if (balanceEntry instanceof BurnedBsqBalanceEntry) {
+            BurnedBsqBalanceEntry burnedBsqBalanceEntry = (BurnedBsqBalanceEntry) balanceEntry;
+            receivedBtc = Optional.empty();
+            burnedBsq = Optional.of(-burnedBsqBalanceEntry.getAmount());
+            type = Optional.of(burnedBsqBalanceEntry.getType());
+        } else {
+            BaseBalanceEntry baseBalanceEntry = (BaseBalanceEntry) balanceEntry;
+            receivedBtc = Optional.of(baseBalanceEntry.getAmount());
+            burnedBsq = Optional.empty();
+            type = Optional.of(baseBalanceEntry.getType());
+        }
+
+        if (price.isEmpty() || receivedBtc.isEmpty()) {
+            receivedBtcAsBsq = Optional.empty();
+        } else {
+            long volume = price.get().getVolumeByAmount(Coin.valueOf(receivedBtc.get())).getValue();
```

Comment:
In a fresh regtest, average BSQ price is 0 and it causes a divide by zero error when used here.

Reply:
You mean the receivedBtc.get() is 0 in that case and getVolumeByAmount throws an exception?

Reply:
Ah I see when the price is 0 the converter methods throw an exception. Unfortunately the ExchangeRate class is from BitcoinJ and lacks a check. I will create a separate PR for moving all those calls to the VolumeUtil and add a check there.